scripts/crew/CrewMember.gd

```diff
@@ -105,13 +129,19 @@ func _on_idle_state_exited() -> void:
 
 func _on_idle_state_physics_processing(delta: float) -> void:
 	if navigation_agent.is_navigation_finished():
-		randomise_target_position()
+		current_state = State.IDLE
+		idle_timer += delta
+		if idle_timer >= idle_time_limit:
+			current_state = State.WALK
+			idle_timer = 0.0
+			randomise_target_position()
 		return
 	
-	if !navigation_agent.is_target_reachable() or (abs(target.x - global_position.x) < 200 and abs(target.y - global_position.y) < 200):
+	if !navigation_agent.is_target_reachable(): # or (abs(target.x - global_position.x) < 200 and abs(target.y - global_position.y) < 200):
```

Comment:
We can bring back the early target recalculation. I just needed to get the `is_navigation_finished()` to fire.